plugins/database/cassandra/tls.go

```diff
@@ -0,0 +1,118 @@
+package cassandra
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+
+	"github.com/hashicorp/vault/sdk/helper/certutil"
+	"github.com/hashicorp/vault/sdk/helper/errutil"
+)
+
+func jsonBundleToTLSConfig(rawJSON string, tlsMinVersion uint16, serverName string, insecureSkipVerify bool) (*tls.Config, error) {
+	certBundle := certutil.CertBundle{}
+	err := json.Unmarshal([]byte(rawJSON), &certBundle)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse JSON: %w", err)
+	}
+
+	if certBundle.IssuingCA != "" && len(certBundle.CAChain) > 0 {
+		return nil, fmt.Errorf("issuing_ca and ca_chain cannot both be specified")
+	}
+	if certBundle.IssuingCA != "" {
+		certBundle.CAChain = []string{certBundle.IssuingCA}
+		certBundle.IssuingCA = ""
+	}
+
+	return toClientTLSConfig(certBundle.Certificate, certBundle.PrivateKey, certBundle.CAChain, tlsMinVersion, serverName, insecureSkipVerify)
+}
+
+func pemBundleToTLSConfig(pemBundle string, tlsMinVersion uint16, serverName string, insecureSkipVerify bool) (*tls.Config, error) {
+	if len(pemBundle) == 0 {
+		return nil, errutil.UserError{Err: "empty pem bundle"}
+	}
+
+	pemBytes := []byte(pemBundle)
+	var pemBlock *pem.Block
+
+	certificate := ""
+	privateKey := ""
+	caChain := []string{}
+
+	count := 0
+	for len(pemBytes) > 0 {
+		count++
+
+		pemBlock, pemBytes = pem.Decode(pemBytes)
+		if pemBlock == nil {
+			return nil, errutil.UserError{Err: "no data found in PEM block"}
+		}
+		blockBytes := pem.EncodeToMemory(pemBlock)
+
+		switch pemBlock.Type {
+		case "CERTIFICATE":
+			// Parse the cert so we know if it's a CA or not
+			cert, err := x509.ParseCertificate(pemBlock.Bytes)
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse certificate: %w", err)
+			}
+			if cert.IsCA {
+				caChain = append(caChain, string(blockBytes))
+				continue
+			}
+
+			// Only one leaf certificate supported
+			if certificate != "" {
+				return nil, errutil.UserError{Err: "multiple leaf certificates not supported"}
+			}
+			certificate = string(blockBytes)
+
+		case "PRIVATE KEY":
+			if privateKey != "" {
+				return nil, errutil.UserError{Err: "multiple private keys not supported"}
+			}
+			privateKey = string(blockBytes)
+		}
```

Comment:
What about PKCS#1 (`RSA PRIVATE KEY`) or EC (`EC PRIVATE KEY`) `Block` types? Should there also be a default for any other type that's considered invalid/unsupported for our use case?

Reply:
Good catch! I'll add `RSA PRIVATE KEY` and `EC PRIVATE KEY` to the `PRIVATE KEY` case.
Re: default: We could either ignore an unsupported block type (current), or error if we find something we don't support. I like the idea of erroring on it to make it clear to the user that what they provided isn't supported rather than failing silently.